apixu-weatherbot1.0.tcl

```diff
@@ -231,69 +233,86 @@ namespace eval weather {
        return $userinfo 
     }
 
-    proc _get_json {location type} {
+    proc _get_weather_data {location type chan} {
         # Fetches JSON data from the apixu API.
         #
         # Args:
         #   location(string) - Users location given.
         #   type(string) - Current weather or forecast called.
         #
         # Returns:
-        #   (string) Returns json data received from the api.
+        #   (dict) Returns json data received from the api and
+        #          converted to a dict.
         #
-        putlog "weather::_get_json was called"
-
-        # First regexp matches any location given by user that is not
-        # a-z, A-Z, number, period, comma or space. This is sanitize
-        # the location string given to catch any illegal characters
-        # before the API call that will return an error.
-        #
-        # Second regexp matches any location give by user that starts
-        # and ends with a-z or A-Z with a : in the between the two. This
-        # is to let strings like "iata:ksjc" to be passed through as the
-        # only illegal character if used the correct way.
-        #
-        # If the first regexp is true and second regexp is false, the
-        # conditional will be true and throw an error for illegal characters.
-        if {[regexp {[^\w., ]} $location] && ![regexp {^[a-zA-Z]+:[a-zA-Z]+$} $location]} {
-            putlog "weather::_get_json caught illegal characters"
-            error "Illegal characters in your location query. Acceptable characters:\
-                   a-z, 0-9, commas, and spaces. Colons only used i.e. 'iata:ksjc'"
+        putlog "weather::_get_weather_datawas called"
+        set query [::http::formatQuery key $::weather::apikey q $location]
+        set url "$::weather::base_url/$type.json?$query"
```

Comment:
perhaps move this to an else of the statement below it for clarity

Reply:
:+1: